services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/asynch/AsynchJobStatusManagerImpl.java

```diff
@@ -173,6 +182,11 @@ public String setComplete(String jobId, AsynchronousResponseBody body)
 			CacheableRequestBody request = (CacheableRequestBody) status.getRequestBody();
 			requestHash = jobHashProvider.getJobHash(request);
 		}
+		// capture the body of the response
+		if (body instanceof S3FileCopyResult || body instanceof S3FileCopyResults) {
```

Comment:
Add interface Snapshotable.
Remove AsynchBody from S3FileCopyResult (it should only be on Results). An S3FileCopyResults should implement Snapshotable.